daemon/daemon.go

```diff
@@ -511,6 +511,14 @@ func (daemon *Daemon) parents(c *container.Container) map[string]*container.Cont
 
 func (daemon *Daemon) registerLink(parent, child *container.Container, alias string) error {
 	fullName := path.Join(parent.Name, alias)
+	aliasChecker := alias
```

Comment:
Aliases should probably follow the same rules as container names, and I think we'd already have a function for that to validate those